test/catalog/catalog_test.cpp

```diff
@@ -274,6 +274,203 @@ TEST_F(CatalogTests, UserIndexTest) {
+  index_builder.SetKeySchema(true_schema);
+  auto index = index_builder.Build();
+
+  EXPECT_TRUE(accessor->SetIndexPointer(idx_oid, index));
+  EXPECT_EQ(common::ManagedPointer(index), accessor->GetIndex(idx_oid));
+  txn_manager_->Commit(txn, transaction::TransactionUtil::EmptyCallback, nullptr);
+
+  // Get an accessor into the database and validate the catalog tables exist
+  // then delete it and verify an invalid OID is now returned for the lookup
+  txn = txn_manager_->BeginTransaction();
+  accessor = catalog_->GetAccessor(txn, db_);
+  EXPECT_NE(accessor, nullptr);
+
+  VerifyTablePresent(*accessor, accessor->GetDefaultNamespace(), "test_table");
+  idx_oid = accessor->GetIndexOid("test_index");
+  EXPECT_NE(idx_oid, catalog::INVALID_INDEX_OID);
+  EXPECT_TRUE(accessor->DropTable(table_oid));
+  VerifyTableAbsent(*accessor, accessor->GetDefaultNamespace(), "test_table");
+  idx_oid = accessor->GetIndexOid("test_index");
+  EXPECT_EQ(idx_oid, catalog::INVALID_INDEX_OID);
+  txn_manager_->Commit(txn, transaction::TransactionUtil::EmptyCallback, nullptr);
+}
+
+/*
+ * Create a user table and index. Drop them all by dropping the namespace using cascading drop logic.
+ */
+// NOLINTNEXTLINE
+TEST_F(CatalogTests, CascadingDropNamespaceTest) {
+  auto txn = txn_manager_->BeginTransaction();
+  auto accessor = catalog_->GetAccessor(txn, db_);
+  EXPECT_NE(accessor, nullptr);
+  auto ns_oid = accessor->CreateNamespace("test_namespace");
+  EXPECT_NE(ns_oid, catalog::INVALID_NAMESPACE_OID);
+  VerifyCatalogTables(*accessor);  // Check visibility to me
+  txn_manager_->Commit(txn, transaction::TransactionUtil::EmptyCallback, nullptr);
+
+  // Create the column definition (no OIDs)
+  std::vector<catalog::Schema::Column> cols;
+  cols.emplace_back("id", type::TypeId::INTEGER, false,
+                    parser::ConstantValueExpression(type::TransientValueFactory::GetNull(type::TypeId::INTEGER)));
+  cols.emplace_back("user_col_1", type::TypeId::INTEGER, false,
+                    parser::ConstantValueExpression(type::TransientValueFactory::GetNull(type::TypeId::INTEGER)));
+  auto tmp_schema = catalog::Schema(cols);
+
+  txn = txn_manager_->BeginTransaction();
+  accessor = catalog_->GetAccessor(txn, db_);
+  EXPECT_NE(accessor, nullptr);
+  auto table_oid = accessor->CreateTable(ns_oid, "test_table", tmp_schema);
+  auto schema = accessor->GetSchema(table_oid);
+  auto table = new storage::SqlTable(&block_store_, schema);
+
+  EXPECT_TRUE(accessor->SetTablePointer(table_oid, table));
+  txn_manager_->Commit(txn, transaction::TransactionUtil::EmptyCallback, nullptr);
+
+  // Create the index
+  txn = txn_manager_->BeginTransaction();
+  accessor = catalog_->GetAccessor(txn, db_);
+  EXPECT_NE(accessor, nullptr);
+  std::vector<catalog::IndexSchema::Column> key_cols{catalog::IndexSchema::Column{
+      "id", type::TypeId::INTEGER, false, parser::ColumnValueExpression(db_, table_oid, schema.GetColumn("id").Oid())}};
```

Comment:
You should have this index be on a table *not* in the dropped namespace in order to test the second pass logic in your code.  As is, this index is really a hidden second test of `DeleteTable`'s cascading logic.

Reply:
Isn't that what the test after this does?